src/udev/udevd.c

```diff
@@ -1759,6 +1759,9 @@ int main(int argc, char *argv[]) {
 
         r = run(fd_ctrl, fd_uevent, cgroup);
 
+        fd_ctrl = -1;
+        fd_uevent = -1;
+
```

Comment:
I think that auto cleanup in ```run()``` will kick in regardless and hence will close the fds. With the one exception, which is failure to allocate manager object right at the beginning of ```run()``` and that is pretty unlikely to happen. Also, marking fds as invalid every time is not a big deal because udevd will exit momentarily anyway.

How about this?

```C
if (r != -ENOMEM) {
          fd_ctrl = -1;
          fd_uevent = -1;
}
```